graph_creator/views.py

```diff
@@ -42,70 +43,105 @@ def get(self, request, *args, **kwargs):
+                    active_students_amount, students_per_country
+            ))
+
+            enthusiast_data = {
+                'latitude': float(received_data.get('latitude')),
+                'longitude': float(received_data.get('longitude')),
+                'platform_name': received_data.get('platform_name'),
+                'platform_url': received_data.get('platform_url'),
+                'students_per_country': students_per_country
+            }
+
+            instance_data.update(enthusiast_data)
+
+        DataStorage.objects.create(**instance_data)
+
     def post(self, request, *args, **kwargs):
         """
         Receive information from the edx-platform and processes it.
-        
+
         If the first time gets information from edx-platform, generates a secret token and
         sends it back to the edx-platform for further data exchange abilities, otherwise
         updates data in the DB with the new incoming information from the edx-platform.
-        
-        `secret_token` when generates is a uuid.UUID in string format.
-        `reverse_token` is a requests. Sends the secret token to edx-platform.
-        
-        Returns http response redirecting to the main page.
+
+        Returns HTTP-response with status 201, that means object (instance data) was successfully created.
         """
 
-        # TODO: make serializer validation for received data
         received_data = self.request.POST
-        active_students_amount = received_data.get('active_students_amount')
-        courses_amount = received_data.get('courses_amount')
-        latitude = received_data.get('latitude')
-        longitude = received_data.get('longitude')
-        platform_name = received_data.get('platform_name')
         platform_url = received_data.get('platform_url')
         secret_token = received_data.get('secret_token')
 
-        # TODO: make creation of data, not updating exist
         if secret_token:
-            DataStorage.objects.filter(secret_token=str(secret_token)).update(
-                active_students_amount=int(active_students_amount),
-                courses_amount=int(courses_amount),
-                latitude=float(latitude),
-                longitude=float(longitude),
-                platform_url=platform_url,
-                platform_name=platform_name,
-            )
-            return HttpResponse(status=200)
+            self.create_instance_data(received_data, secret_token)
 
         else:
             secret_token = uuid.uuid4().hex
-            DataStorage.objects.create(
-                active_students_amount=int(active_students_amount),
-                courses_amount=int(courses_amount),
-                latitude=float(latitude),
-                longitude=float(longitude),
-                platform_name=platform_name,
-                platform_url=platform_url,
-                secret_token=secret_token
-            )
+            self.create_instance_data(received_data, secret_token)
```

Comment:
I did it, total is I cant`s do that, because if secret_token is empty, I need to make additional logic (dispatch token to edX`s DB). That means I need extend create_instance_data(), that makes obscurity as well as I save initial secret token in special variable.